### PR TITLE
Add server config to disable user registered agents

### PR DESCRIPTION
**Pull request by 6543**
As user-/org-agents allow normal (non instance admins) to extract secrets if they know how from the repos they are admin at, this is an hardening option.

normally you already trust users who are admin but in edge cases you want to forbid them to register own agents ...

### CODE REVIEW

**Comment by 6543** [comment]
It looked nicer but the only real answer is that i did not come up witj a combined name for the var

File: server/api/agent.go [hunk 1]
```diff
@@ -250,6 +250,10 @@ func PostOrgAgent(c *gin.Context) {
 	_store := store.FromContext(c)
 	user := session.User(c)
 
+	if !user.Admin && server.Config.Agent.DisableNonAdminAgents {
+		c.String(http.StatusForbidden, "creation of user-/org-agents was disabled by the woodpecker admin")
```
Comment (6543) [inline server/api/agent.go:254]: bad fail!

Comment (6543) [reply server/api/agent.go:254]: -> https://github.com/woodpecker-ci/woodpecker/pull/4206/commits/049d78ead9ab18da446a73c496c4388e0da32520

File: cmd/server/flags.go [hunk 2]
```diff
@@ -193,6 +193,11 @@ var flags = append([]cli.Flag{
 		Name:  "agent-secret",
 		Usage: "server-agent shared password",
 	},
+	&cli.BoolFlag{
+		Sources: cli.EnvVars("WOODPECKER_DISABLE_USER_AGENT_REGISTRATION"),
```
[the remark this answers is not in the document]
Comment (6543) [reply cmd/server/flags.go:197]: well but that would imply that old agents are now invalid ... witch they are not and that is intended

File: server/api/agent.go [hunk 3]
```diff
@@ -250,6 +250,11 @@ func PostOrgAgent(c *gin.Context) {
 	_store := store.FromContext(c)
 	user := session.User(c)
 
+	if !user.Admin && server.Config.Agent.DisableNonAdminAgents {
```
[the remark this answers is not in the document]
Comment (6543) [reply server/api/agent.go:253]: yes but he then would also manually emulate how org agents behave ... I want old registered once still valid and be able to be managed too...

File: server/config.go [hunk 4]
```diff
@@ -54,6 +54,9 @@ var Config = struct {
 		CustomCSSFile       string
 		CustomJsFile        string
 	}
+	Agent struct {
+		DisableNonAdminAgents bool
```
Comment (6543) [inline server/config.go:58]:
```suggestion
		DisableNonAdminAgentRegistration bool
```

File: cmd/server/flags.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (6543) [reply cmd/server/flags.go:197]: if the admin want to get rid of it, just delete them in the admin view ...

[the remark this answers is not in the document]
Comment (6543) [reply cmd/server/flags.go:197]: ... and if they should not exist he can set this before upgrading to v3.0.0 ,,,

File: docs/docs/30-administration/10-server-config.md [hunk 5]
```diff
@@ -54,6 +54,20 @@ Use the `WOODPECKER_REPO_OWNERS` variable to filter which GitHub user's repos sh
...
+By default non instance admins can create new agents for their user-/org-repos they have admin access to.
+If an instance admin don't want this feature enabled, they can disable the registration of new one.
+
+:::note
+You should set this option if you have, for example,
+global secrets and don't trust your users to create a rogue agent and pipeline for secret extraction.
+:::
+
+```env
+WOODPECKER_DISABLE_USER_AGENT_REGISTRATION=true
```
[the remark this answers is not in the document]
Comment (6543) [reply docs/docs/30-administration/10-server-config.md:68]: done

**Comment by 6543** [comment]
well this would allow instance admins to still create org agents ... but not normal users

there is no much downside in doing so ... this is the same argument i wana have repo agents.

It is easier in therms of UX for the user and do not add much code on our side to maintain

**Comment by anbraten** [comment]
> well this would allow instance admins to still create org agents ... but not normal users

Would probably help to ask: What would be the argument for an instance admin to disable user agents in general.

**Comment by 6543** [comment]
I wrote that argument also into the docs

**Comment by 6543** [comment]
https://github.com/woodpecker-ci/woodpecker/pull/4206/files#diff-797f56c545a9d6d7c5a0a660ecf679cef504d2538941b85876458b60a6e7933fR63

**Comment by 6543** [comment]
now the flack disable the function & ui alltogether

File: docs/docs/30-administration/10-server-config.md [hunk 6]
```diff
@@ -54,6 +54,20 @@ Use the `WOODPECKER_REPO_OWNERS` variable to filter which GitHub user's repos sh
 WOODPECKER_REPO_OWNERS=my_company,my_company_oss_github_user
 ```
 
+## Disallow normal users to create agents
+
+By default, users can create new agents for their repos they have admin access to.
+If an instance admin doesn't want this feature enabled, they can disable the registration of new one.
```
Comment (6543) [inline docs/docs/30-administration/10-server-config.md:60]:
```suggestion
If an instance admin doesn't want this feature enabled, they can disable the API and hide the Web UI elements.
```

File: web/src/compositions/useConfig.ts [hunk 7]
```diff
@@ -18,4 +19,5 @@ export default () => ({
   csrf: window.WOODPECKER_CSRF ?? null,
   rootPath: window.WOODPECKER_ROOT_PATH ?? '',
   enableSwagger: window.WOODPECKER_ENABLE_SWAGGER === true || false,
+  userRegisteredAgents: window.WOODPECKER_USER_REGISTERED_AGENTS || window.WOODPECKER_USER?.admin,
```
Comment (6543) [inline web/src/compositions/useConfig.ts:22]:
```suggestion
  userRegisteredAgents: window.WOODPECKER_USER_REGISTERED_AGENTS || false,
```

**Pull request merged**